client/ayon_core/plugins/load/export_otio.py

```diff
@@ -0,0 +1,187 @@
+from collections import defaultdict
+
+from qtpy import QtWidgets, QtCore, QtGui
+
+from ayon_core.client import (
+    get_representations,
+    get_version_by_id
+)
+from ayon_core.pipeline import load, Anatomy
+from ayon_core import resources, style
+from ayon_core.pipeline.editorial import export_otio
+from ayon_core.pipeline.load import get_representation_path_with_anatomy
+from ayon_core.lib import run_subprocess
+
+
+class ExportOTIO(load.SubsetLoaderPlugin):
+    """Export selected versions to OpenTimelineIO."""
+
+    is_multiple_contexts_compatible = True
+    sequence_splitter = "__sequence_splitter__"
+
+    representations = ["*"]
+    families = ["*"]
+    tool_names = ["library_loader"]
+
+    label = "Export OTIO"
+    order = 35
+    icon = "save"
+    color = "#d8d8d8"
+
+    def load(self, contexts, name=None, namespace=None, options=None):
+        try:
+            dialog = ExportOTIOOptionsDialog(contexts, self.log)
+            dialog.exec_()
+        except Exception:
+            self.log.error("Failed to export OTIO.", exc_info=True)
+
+
+class ExportOTIOOptionsDialog(QtWidgets.QDialog):
+    """Dialog to select template where to deliver selected representations."""
+
+    def __init__(self, contexts, log=None, parent=None):
+        super(ExportOTIOOptionsDialog, self).__init__(parent=parent)
+
+        self.setWindowTitle("AYON - Export OTIO")
+        icon = QtGui.QIcon(resources.get_ayon_icon_filepath())
+        self.setWindowIcon(icon)
+
+        self.setWindowFlags(
+            QtCore.Qt.WindowStaysOnTopHint
+            | QtCore.Qt.WindowCloseButtonHint
+            | QtCore.Qt.WindowMinimizeButtonHint
+        )
+
+        self.setStyleSheet(style.load_stylesheet())
+
+        input_widget = QtWidgets.QWidget(self)
+        input_layout = QtWidgets.QGridLayout(input_widget)
+
+        self._project_name = contexts[0]["project"]["name"]
+
+        representations_by_version_id = defaultdict(list)
+        self._version_by_representation_id = {}
+        all_representation_names = set()
+        self._version_path_by_id = {}
+        for context in contexts:
+            version_id = context["version"]["_id"]
+            version = get_version_by_id(self._project_name, version_id)
+            representations = list(get_representations(
+                self._project_name, version_ids=[version_id]
+            ))
+            representations_by_version_id[version_id] = representations
+
+            for representation in representations:
+                all_representation_names.add(representation["name"])
+                id = representation["_id"]
+                self._version_by_representation_id[id] = version
+
+            self._version_path_by_id[version_id] = "{}/{}/{}/v{:03d}".format(
+                representations[0]["context"]["hierarchy"],
+                representations[0]["context"]["asset"],
+                representations[0]["context"]["subset"],
```

Comment:
You should not expect that representation context has `"asset"`, `"subset"` and `"hierarchy"`.

Reply:
Ok, any suggestion how to get the hierarchy path of a version?